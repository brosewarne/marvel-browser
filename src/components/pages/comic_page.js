import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Paper from 'material-ui/Paper';

import { ItemImage } from '../items/item_image';

import './pages.css';

/**
 * Render the ComicPage, showing a profile of the selected comic
 * @param {Object} props - The set of props for the ComicPage - passed from the route
 * @returns {Node} - The rendered ComicPage
 * @constructor
 */
export const _ComicPage = (props) => {
    const {
        title, description
    } = props.item;

    return (
        <div>
            <content className="pageContainer">
                <Paper elevation={12} className="itemProfile">
                    <h1 className="itemProfileTitle">{title}</h1>
                    <ItemImage item={props.item} imgSize="portrait_uncanny" />
                    <p>{description || 'No comic description available'}</p>
                </Paper>
            </content>
        </div>
    );
};

_ComicPage.propTypes = {
    item: PropTypes.object.isRequired
};

/**
 * If item is not supplied from props, get it from the app state
 * @param {Object} state - The current app state
 * @param {Object} props - The incoming set of props
 * @returns {Object} - Empty id item is already in props, otherwise, the mapped item from app state
 */
const mapStateToProps = (state, props) => {
    return props.item ? {} : {
        item: state.routing.location.state.data
    };
};

export const ComicPage = connect(mapStateToProps)(_ComicPage);
