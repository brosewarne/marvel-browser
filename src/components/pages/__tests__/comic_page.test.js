import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { _ComicPage } from '../comic_page';

configure({ adapter: new Adapter() });

describe('<_ComicPage />', () => {
    const mockComic = {
        title: 'Comic Title',
        description: 'Comic Description',
        thumbnail: {
            path: '/a/href/to/an/image',
            extension: 'png'
        },
        urls: [
            {
                url: '/aURL',
                type: 'wiki'
            }
        ]
    };
    it('renders correctly', () => {
        const wrapper = shallow(
            <_ComicPage item={mockComic} />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
