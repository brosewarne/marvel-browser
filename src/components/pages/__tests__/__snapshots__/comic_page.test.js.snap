// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<_ComicPage /> renders correctly 1`] = `
<div>
  <content
    className="pageContainer"
  >
    <WithStyles(Paper)
      className="itemProfile"
      elevation={12}
    >
      <h1
        className="itemProfileTitle"
      >
        Comic Title
      </h1>
      <Component
        imgSize="portrait_uncanny"
        item={
          Object {
            "description": "Comic Description",
            "thumbnail": Object {
              "extension": "png",
              "path": "/a/href/to/an/image",
            },
            "title": "Comic Title",
            "urls": Array [
              Object {
                "type": "wiki",
                "url": "/aURL",
              },
            ],
          }
        }
      />
      <p>
        Comic Description
      </p>
    </WithStyles(Paper)>
  </content>
</div>
`;
